src/ui/ResultFolding/ResultFolding.ts

```diff
@@ -302,51 +300,40 @@ export class ResultFolding extends Component {
     window.scrollTo(0, new Win(window).scrollY() + resultElem.getBoundingClientRect().top);
   }
 
-  private displayThoseResults(results: IQueryResult[]): Promise<boolean> {
+  private async displayThoseResults(results: IQueryResult[]): Promise<boolean> {
     const childResultsPromises = _.map(results, result => {
       return this.renderChildResult(result);
     });
 
-    return Promise.all(childResultsPromises).then((childsToAppend: HTMLElement[]) => {
-      $$(this.results).empty();
-      _.each(childsToAppend, oneChild => {
-        this.results.appendChild(oneChild);
-      });
-      return true;
+    const childsToAppend: HTMLElement[] = await Promise.all(childResultsPromises);
+    $$(this.results).empty();
+    _.each(childsToAppend, oneChild => {
+      this.results.appendChild(oneChild);
     });
+    return true;
   }
 
-  private renderChildResult(childResult: IQueryResult): Promise<HTMLElement> {
+  private async renderChildResult(childResult: IQueryResult): Promise<HTMLElement> {
     QueryUtils.setStateObjectOnQueryResult(this.queryStateModel.get(), childResult);
     QueryUtils.setSearchInterfaceObjectOnQueryResult(this.searchInterface, childResult);
 
-    return this.options.resultTemplate
-      .instantiateToElement(childResult, {
-        wrapInDiv: false,
-        checkCondition: false,
-        responsiveComponents: this.searchInterface.responsiveComponents
-      })
-      .then((oneChild: HTMLElement) => {
-        $$(oneChild).addClass('coveo-result-folding-child-result');
-
-        $$(oneChild).toggleClass('coveo-normal-child-result', !this.showingMoreResults);
-        $$(oneChild).toggleClass('coveo-expanded-child-result', this.showingMoreResults);
-        return this.autoCreateComponentsInsideResult(oneChild, childResult).initResult.then(() => {
-          return oneChild;
-        });
-      });
+    const oneChild: HTMLElement = await this.options.resultTemplate.instantiateToElement(childResult, {
+      wrapInDiv: false,
+      checkCondition: false,
+      responsiveComponents: this.searchInterface.responsiveComponents
+    });
+
+    $$(oneChild).addClass('coveo-result-folding-child-result');
+    $$(oneChild).toggleClass('coveo-normal-child-result', !this.showingMoreResults);
+    $$(oneChild).toggleClass('coveo-expanded-child-result', this.showingMoreResults);
+
+    await this.autoCreateComponentsInsideResult(oneChild, childResult).initResult;
+    return oneChild;
   }
 
   private autoCreateComponentsInsideResult(element: HTMLElement, result: IQueryResult): IInitResult {
```

Comment:
Good catch !

I would remove that method completely however, since it's not doing much anymore, and simply call `Initialization.automaticallyCreateComponentsInsideResult(element, result)` directly where needed.